cinema/views.py

```diff
@@ -56,3 +74,43 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        if self.action == "list":
+            date = self.request.query_params.get("date")
+            movies = self.request.query_params.get("movie")
+
+            if date:
+                queryset = queryset.filter(show_time__date=date,)
+
+            if movies:
+                movies_ids = [int(str_id) for str_id in movies.split(",")]
+                queryset = queryset.filter(movie_id__in=movies_ids)
+
+            queryset = (
+                queryset
+                .select_related("cinema_hall",)
+                .prefetch_related("tickets")
+            ).order_by("id")
+
+            for obj in queryset:
+                obj.tickets_available = (
+                    obj.cinema_hall.capacity - len(obj.tickets.all())
+                )
+        return queryset
+
+
+class OrderSessionViewSet(viewsets.ModelViewSet):
```

Comment:
The class `OrderSessionViewSet` might be incorrectly named. It is likely intended to be `OrderViewSet` based on the context of handling orders. Verify the intended name and update it for consistency across the application.